File: DailyLeetcoding/792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        // int start = 0, end = nums.length - 1;
        // while(start <= end){
        //     int mid = start + (end - start) / 2;
        //     if(nums[mid] == target){
        //         return mid;
        //     }
        //     if(nums[mid] > target){
        //         end = mid - 1;
        //     } else {
        //         start = mid + 1;
        //     }
        // }
        // return -1;
        return binSearchRec(nums, target, 0, nums.length - 1);
    }
    static int binSearchRec(int[] nums, int target, int start, int end){
        // base case
        if(start > end){
            return -1;
        }

        int mid = start + (end - start) / 2;
        if(nums[mid] == target){
            return mid;
        }
        if(nums[mid] > target){
            return binSearchRec(nums, target, start, mid - 1);
        }
        return binSearchRec(nums, target, mid + 1, end);       
    }
}

File: DailyLeetcoding/792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        def binSearch(nums, target, start, end):

            if start <= end:
                middle = (start + end) // 2

                if nums[middle] == target:
                    print(middle)
                    return middle
                elif nums[middle] < target:
                    return binSearch(nums, target, middle + 1, end)
                else: 
                    return binSearch(nums, target, start, middle - 1)
            else:
                return -1

        return binSearch(nums, target, 0, len(nums) - 1)